code/items/equipments/entities/HealthstationEntity.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+
+using Sandbox;
+
+using TTTReborn.Player;
+
+namespace TTTReborn.Items
+{
+    [Library("ttt_healthstation_ent"), Hammer.Skip]
+    public partial class HealthstationEntity : Prop, IUse
+    {
+        [Net]
+        public float StoredHealth { get; set; } = 200; //This number technically has to be a float for the methods to work, but it should stay a whole number the entire time.
+
+        private string ModelPath => "models/entities/healthstation.vmdl";
+
+        private RealTimeUntil NextHeal = 0;
+
+        private const int HealAmount = 1;
+        private const int HealFrequency = 2; //seconds
+        private const int DelayIfFailed = 2; //Multiplied by HealFrequency if HealthPlayer returns false
+
+        public override void Spawn()
+        {
+            base.Spawn();
+
+            SetModel(ModelPath);
+            SetupPhysicsFromModel(PhysicsMotionType.Dynamic);
+        }
+
+        private bool HealPlayer(TTTPlayer player)
+        {
+            if (StoredHealth > 0)
+            {
+                float healthNeeded = player.MaxHealth - player.Health;
+
+                if (healthNeeded > 0)
+                {
+                    var healAmount = Math.Min(HealAmount, healthNeeded);
+
+                    player.SetHealth(player.Health + healAmount);
+                    StoredHealth -= healAmount;
+
+                    //Event.Run("tttreborn.healthstation.healed", player, healAmount); //Would have liked to pass `this` as well, but apparently events are capped at 2 parameters?
```

Comment:
Is this still needed?

Reply:
Not specifically. Waiting at the moment to see if we'll be able to pass multiple parameters off an event, or have to make it a struct. I say leave it in for the moment.